From fa9f15e3be811bd1e55938fa92a110f9d389342d Mon Sep 17 00:00:00 2001
From: Joe Lawrence <Joe.Lawrence@stratus.com>
Date: Thu, 24 Jan 2013 17:26:43 -0500
Subject: [PATCH] Support out-of-tree and updated drivers in initramfs creation

The find_kernel_modules_by_path function shouldn't filter out modules not
located underneath the /lib/modules/ver/kernel directory as out-of-tree
drivers may reside in /lib/modules/ver/extra and updated drivers in
/lib/modules/ver/updates.

Signed-off-by: Joe Lawrence <joe.lawrence@stratus.com>
---
 dracut-functions.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dracut-functions.sh b/dracut-functions.sh
index e23db76..f7e3ec8 100755
--- a/dracut-functions.sh
+++ b/dracut-functions.sh
@@ -1502,7 +1502,7 @@ find_kernel_modules_by_path () (
         _OLDIFS=$IFS
         IFS=:
         while read a rest; do
-            [[ $a = kernel*/$1/* ]] || continue
+            [[ $a = */$1/* ]] || continue
             echo $srcmods/$a
         done < $srcmods/modules.dep
         IFS=$_OLDIFS
